Automattic/jetpack
# Boost: Make sure source provider URLs are unique

## Proposed changes:

- Remove core_front_page provider and code removing it. We added this when working on Cornerstone Pages, and the logic was behind a feature flag. There's no longer a feature flag, so it's okay to remove the source provider and the code that was previously removing it.
- Update critical css UI to consider cornerstone pages as important and show an error if one of them fails. core_front_page is no longer an important page as it no longer exists as a provider. This partially addresses https://github.com/Automattic/jetpack/issues/40391.
- Remove WP Core Provider URLs from list, if they are present in Cornerstone Pages list;
- Remove possible duplicate blog page, if it was in the cornerstone pages list when a static page was not set, but a blog page was set.
- Fix bug that made "Load default pages" button clickable after loading defaults and saving. This happens when Yoast had the home page as a cornerstone page and it was also in Boost's cornerstone pages list.

### Other information:


## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

- Setup Boost (free);
- You can use Boost companion, to easily check the list of source providers by opening http://jetpack-boost.test/wp-json/jetpack-boost/v1/list-source-providers
- Add some pages, set a static page, no blog page, add cornerstone pages. In other words - play around with the cornerstone pages and see if the list of source providers behaves correctly.
